lib/improver/cli/__init__.py

```diff
@@ -356,18 +358,25 @@ def execute_command(dispatcher, prog_name, *args,
             # process nested commands recursively
             arg = execute_command(dispatcher, prog_name, *arg,
                                   verbose=verbose, dry_run=dry_run)
-        if not isinstance(arg, str):
+        if isinstance(arg, pathlib.PurePath):
+            arg = str(arg)
+        elif not isinstance(arg, str):
             arg = ObjectAsStr(arg)
         args[i] = arg
 
+    if verbose or dry_run:
+        print(" ".join([shlex.quote(x) for x in (prog_name, *args)]))
     if dry_run:
-        result = args
-    else:
-        result = dispatcher(prog_name, *args)
+        return args
 
-    if verbose or dry_run:
-        print(prog_name, *args, ' -> ', ObjectAsStr.obj_to_name(result))
+    try:
+        result = dispatcher(prog_name, *args)
+    except Exception as exc:
+        print(exc)
+        raise exc
```

Comment:
I guess we could skip printing the exception here (it will go to stderr anyway) and simplify this to just:
```python
    result = dispatcher(prog_name, *args)
```